src/Services/src/DetailedResult.cs

```diff
@@ -19,7 +19,7 @@ public DetailedResult(bool succeeded, string explanation = null, CmdResult cmdDe
 
         public bool Succeeded { get; set; }
         public string Explanation { get; set; }
-        public CmdResult CmdDetails { get; }
+        public CommandResult CmdDetails { get; }
```

Comment:
var name is effectively semantically the same as type name but yet not the same; perhaps name variable `CmdResult` or `CommandResult`?